blockchain/src/state/read/model.rs

```diff
@@ -162,6 +162,29 @@ pub trait StateQueries {
         Ok(bytes.map(Bytes))
     }
 
+    /// Queries the blockchain state version corresponding with block `height` for the value of a
+    /// single EVM storage slot `index` at `account`.
+    fn evm_storage_at(
+        &self,
+        evm_storage: &impl StorageTrieRepository,
+        account: Address,
+        index: U256,
+        height: BlockHeight,
+    ) -> Result<U256, state::Error> {
+        let resolver = self.resolver_at(height)?;
+
+        // Read account info to get the storage root
+        let evm_db = ResolverBackedDB::new(evm_storage, &resolver, &(), 0);
```

Comment:
Nit: could use `height` here instead of `0`. It doesn't matter because the block height is not relevant to the lookup we are doing on the EVM state here, but it might be nice to pass it in properly anyway.